fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -197,6 +205,16 @@ class Agile implements AgilityContext {
         private boolean committingNow = false;
         private long prevCommitCheckTime;
         private boolean closed = false;
+        private volatile Function<FDBRecordContext, CompletableFuture<Boolean>> contextPreconditions = context -> AsyncUtil.READY_TRUE;
```

Comment:
I only just started learning about `VarHandle`, but according to the JEP, the volatile may be unnecessary: https://openjdk.org/jeps/193

Reply:
I have used only `Atomic*FieldUpdater` so far too. 

> A field need not be marked as volatile for an associated VarHandle to perform volatile access. In effect, the volatile modifier, if present, is ignored.

I am not sure how to interpret this. The `VarHandle` access doesn't need it but what about the plain read? My guess is that the `volatile` is not needed too. The old JVMs were adding `mfence` instruction only after a write. Most likely the `VarHanlde` codegen does something similar even when working on a plain field. Ignoring the `volatile` probably means that they put the memory fence anyway. But that may be a different type of memory fence (or the ARM acquire/release). 
This is why I left the `volatile`. It is either ignored or it should be there for the plain read.